backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/utils/VdsCpuUnitPinningHelper.java

```diff
@@ -145,7 +146,7 @@ private void previewPinOfPendingExclusiveCpus(List<VdsCpuUnit> cpuTopology, Map<
                 int coreThreads = freeCpusInCore.size();
                 while (coreThreads >= vm.getThreadsPerCpu() &&
                         cpusToBeAllocated.size() < vm.getNumOfCpus() &&
-                        cpuCounter / amountOfCpusPerNuma < highestAmountOfvNumaNodesInSocket) {
+                        (cpuCounter + vm.getThreadsPerCpu()) / amountOfCpusPerNuma <= highestAmountOfvNumaNodesInSocket) {
```

Comment:
this seems correct but maybe we can simplify this code when thinking about sockets rather than cores:
if we have a physical socket that can cope with 1 vnuma node (in terms of memory) and each vsocket contains 2 vnuma nodes - then we can say the physical socket cannot cope with any vsocket, right?
similarly, if the physical socket can cope with 4 numa nodes (in terms of memory) and again each vsocket contains 2 vnuma nodes - then we can say this physical socket cannot cope with more than 2 vsockets (regardless of the free cpus in it)
that way the reminder is not only for partial sockets but also for full sockets that we don't have enough memory for, does it make sense?

Reply:
i think it does, but instead of simplifying i think it's more complex.
instead of using this simple addition to the code i will need to check how many vnuma are in vsocket.
we did the whole logic bottom-up, looking now from "above" is just switching to top-down.

Reply:
yes, it will require a bit more changes in the code but the code will be more explicit and can reduce the amount of math here :)